admin-dev/themes/new-theme/js/pages/product/edit/product-suppliers-manager.js

```diff
@@ -0,0 +1,213 @@
+
+    this.suppliers.forEach((supplier) => {
+      if (supplier.deleted) {
+        return;
+      }
+
+      const productSupplierRow = productSupplierRowPrototype
+        .replace(new RegExp(ProductMap.supplierIdPlaceholder, 'g'), supplier.supplierId)
+        .replace(new RegExp(ProductMap.supplierNamePlaceholder, 'g'), supplier.supplierName);
+
+      this.$productSuppliersTBody.append(productSupplierRow);
+      // Fill inputs
+      $(ProductMap.suppliersSupplierIdInput(supplier.supplierId)).val(supplier.supplierId);
+      $(ProductMap.suppliersProductSupplierIdInput(supplier.supplierId)).val(supplier.productSupplierId);
+      $(ProductMap.suppliersSupplierNameInput(supplier.supplierId)).val(supplier.supplierName);
+      $(ProductMap.suppliersProductSupplierReferenceInput(supplier.supplierId)).val(supplier.reference);
+      $(ProductMap.suppliersProductSupplierPriceInput(supplier.supplierId)).val(supplier.price);
+      $(ProductMap.suppliersProductSupplierCurrencyIdInput(supplier.supplierId)).val(supplier.currencyId);
+    });
+  }
+
+  getSelectedSuppliers() {
+    const selectedSuppliers = [];
+    this.$supplierSelectionBlock.find('input:checked').each((index, input) => {
+      selectedSuppliers.push({
+        name: input.dataset.label,
+        id: input.value,
+      });
+    });
+
+    return selectedSuppliers;
+  }
+
+  refreshDefaultSupplierBlock() {
+    const suppliers = this.getSelectedSuppliers();
+
+    if (suppliers.length === 0) {
+      this.$defaultSuppliersSelectionBlock.find('input').prop('checked', false);
+      this.hideDefaultSuppliers();
+
+      return;
+    }
+
+    this.showDefaultSuppliers();
+    const selectedSupplierIds = suppliers.map((supplier) => supplier.id);
+
+    this.$defaultSuppliersSelectionBlock.find('input').each((key, input) => {
+      const isValid = selectedSupplierIds.includes(input.value);
+
+      if (!isValid && input.checked) {
+        input.checked = false;
+      }
+      input.disabled = !isValid;
+    });
+
+    if (this.$defaultSuppliersSelectionBlock.find('input:checked').length === 0) {
+      this.checkFirstAvailableDefaultSupplier(selectedSupplierIds);
+    }
+  }
+
+  hideDefaultSuppliers() {
+    this.$defaultSuppliersSelectionBlock.addClass('d-none');
+  }
+
+  showDefaultSuppliers() {
+    this.$defaultSuppliersSelectionBlock.removeClass('d-none');
+  }
+
+  checkFirstAvailableDefaultSupplier(selectedSupplierIds) {
+    const firstSupplierId = selectedSupplierIds[0];
+    this.$defaultSuppliersSelectionBlock.find(`input[value="${firstSupplierId}"]`).prop('checked', true);
+  }
+
+  showTable() {
+    this.$productSuppliersTable.removeClass('d-none');
+  }
+
+  hideTable() {
+    this.$productSuppliersTable.addClass('d-none');
+  }
+
+  collectCurrentSuppliers() {
+    this.getSelectedSuppliers().forEach((supplier) => {
+      this.suppliers[supplier.id] = {
+        supplierId: $(ProductMap.suppliersSupplierIdInput(supplier.id)).val(),
```

Comment:
```suggestion
        supplierId: supplier.id,
```
This one doesn't need to be fetched from DOM